src/main/resources/db/migration/V1_117__update_print_files_to_clob_procedure.sql

```diff
@@ -0,0 +1,168 @@
+              				THEN NOW()::date - 1
+               				ELSE NOW()
+              		END;
+
+    -- delete redundant letter requests
+	CALL juror_mod.delete_printfiles();
+
+	FOR forms IN
+		SELECT	fr.form_type,
+				fr.max_rec_len::integer
+		FROM juror_mod.t_form_attr fr
+
+		loop
+			CALL juror_mod.write_printfiles(forms.form_type, forms.max_rec_len, v_ext_date, p_limit, v_count);
+			IF v_count >= p_limit THEN
+				EXIT;
+			END IF;
+
+		END LOOP;
+
+END;
+$procedure$
+;
+
+-- DROP PROCEDURE juror_mod.write_printfiles(in varchar, in int4, in date, in int4, inout int4);
+
+CREATE OR REPLACE PROCEDURE juror_mod.write_printfiles(IN p_form_type character varying, IN p_max_rec_len integer, IN p_ext_date date, IN p_limit integer, INOUT p_count integer)
+ LANGUAGE plpgsql
+AS $procedure$
+DECLARE
+	form_details RECORD;
+	v_filename VARCHAR(50);
+	v_header VARCHAR(50);
+	v_row_id INTEGER;
+	v_count INTEGER:=1;
+	v_strpos INTEGER:=1;
+	v_max_loop_count float;
+	v_data text:='';
+	v_max_data_len float;
+	v_detail_rec text;
+	v_revision int8;
+   	v_text_var1 VARCHAR(255);
+   	v_text_var2 VARCHAR(255);
+   	v_text_var3 VARCHAR(255);
+	v_print_msg VARCHAR(255);
+
+begin
+	-- Identify the next sequence number
+	SELECT  NEXTVAL('juror_mod.print_file_count') INTO v_revision;
+	v_filename := 'JURY'||LPAD(v_revision::VARCHAR(22),4,'0')||'01.0001';
+
+	FOR form_details IN
+		SELECT 	ROW_NUMBER() OVER () as counter,
+				bpd.id,
+				REPLACE(REPLACE(bpd.detail_rec,CHR(10),' '),CHR(13),' ') detail_rec
+		FROM juror_mod.bulk_print_data bpd
+		WHERE bpd.form_type = p_form_type
+		AND COALESCE(bpd.extracted_flag,'N') = 'N'
+		AND DATE(bpd.creation_date) <= p_ext_date
+	LOOP
+
+		IF p_count >= p_limit then
+			-- Create the record
+			INSERT INTO juror_mod.content_store(request_id,document_id,file_type,data)
+			VALUES (NEXTVAL('juror_mod.content_store_seq'),v_filename,'PRINT',v_data);
+			COMMIT;
+
+			-- setting new file name as last file has filled up to the limit
+			SELECT  NEXTVAL('juror_mod.print_file_count') INTO v_revision;
+			v_filename := 'JURY'||LPAD(v_revision::VARCHAR(22),4,'0')||'01.0001';
+
+			-- erasing data for new file
+			v_data := '';
+			p_count := 0;
+
+
+		END IF;
+
+		-- Form the data column details first
+		SELECT 	form_details.id,
+				RPAD('   ' ||RPAD(p_form_type,16)||LPAD(form_details.counter::VARCHAR(20),6,'0')||LPAD(form_details.counter::VARCHAR(20),6,'0')||'50'||LPAD(p_max_rec_len::VARCHAR(20),8,'0'),256,' '),
+				form_details.detail_rec
+		INTO v_row_id, v_header, v_detail_rec;
+
+		SELECT LENGTH(form_details.detail_rec)
+		INTO v_max_data_len;
+
+		-- round up the division to the nearest whole number - e.g. 1.4 would equate to 2
+		SELECT CEIL(v_max_data_len/p_max_rec_len)
+		INTO v_max_loop_count;
+
+		-- If data is empty then the first line will be the header plus newline otherwise it will be the current data plus header plus newline
+		v_data := v_data||v_header||CHR(10);
```

Comment:
We appear to be doing this every time we process a row from bulk_print_data. Won't that mean that we add the header row every time we add a row from bulk_print_data? We need the header once at the top of each file.